api/v1alpha1/jwt_types.go

```diff
@@ -81,3 +85,28 @@ type ClaimToHeader struct {
 	// to separate the JSON name path.
 	Claim string `json:"claim"`
 }
+
+// JWTHeaderExtractor defines an HTTP header location to extract JWT token
+type JWTHeaderExtractor struct {
+	// Name is the HTTP header name to retrieve the token
+	//
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// ValuePrefix is the prefix that should be stripped before extracting the token.
+	// The format would be used by Envoy like "{ValuePrefix}<TOKEN>".
+	// For example, "Authorization: Bearer <TOKEN>", then the ValuePrefix="Bearer " with a space at the end.
+	ValuePrefix *string `json:"valuePrefix,omitempty"`
+}
+
+// JWTExtractor defines a custom JWT token extraction from HTTP request.
+type JWTExtractor struct {
```

Comment:
what happens when one of these fields is set ? does Envoy no longer peek into `Authorization: Bearer` ?

Reply:
as per my observation, yes, it won't check to `Authorization` HTTP header, reference: https://github.com/envoyproxy/envoy/tree/main/api/envoy/config/filter/http/jwt_authn/v2alpha#the-locations-to-extract-jwt